### PR TITLE
SCRUM-9 - Feature/loaders

### PR DESCRIPTION
**Pull request by Annel18**

### CODE REVIEW

File: src/utils/loaders/getCityFeed.test.js [hunk 1]
```diff
@@ -1,4 +1,27 @@
...
+    })
+  )
+
+  beforeEach(() => {
+    fetch.mockClear()
+  })
+
+  it('should call fetch with the correct URL', async () => {
+    const token = 'mockToken'
+    const expectedURL = `https://api.waqi.info/feed/here/?token=${token}`
```
Comment (rasimsen) [inline src/utils/loaders/getCityFeed.test.js:17]: url from ENV file, please. 

Please note: we are gonna use axios, we are gonna replace for axios next sprint

File: src/utils/loaders/getCityFeed.test.js [hunk 2]
```diff
@@ -1,4 +1,27 @@
...
+    await getCityFeed(token)
+    expect(fetch).toHaveBeenCalledWith(expectedURL)
+  })
+
+  it('should return JSON data from the response', async () => {
+    const token = 'mockToken'
+    const result = await getCityFeed(token)
+    expect(result).toEqual({ mockData: 'mockValue' })
```
Comment (rasimsen) [inline src/utils/loaders/getCityFeed.test.js:25]: can you test with some data that close to real data, at least structure wise

File: src/utils/loaders/getCityFeed.test.js [hunk 3]
```diff
@@ -1,4 +1,27 @@
...
+    Promise.resolve({
+      json: () => Promise.resolve({ mockData: 'mockValue' }),
+    })
+  )
+
+  beforeEach(() => {
+    fetch.mockClear()
+  })
+
+  it('should call fetch with the correct URL', async () => {
```
Comment (rasimsen) [inline src/utils/loaders/getCityFeed.test.js:15]: can you add unit tests also for exceptions ! for the worse case scenerios

File: src/utils/loaders/getCityFeed.test.js [hunk 4]
```diff
@@ -1,4 +1,27 @@
...
+    const token = 'mockToken'
```
Comment (rasimsen) [inline src/utils/loaders/getCityFeed.test.js:16]: what if, token is not set!

File: src/utils/loaders/getCityFeed.js [hunk 5]
```diff
@@ -1,7 +1,4 @@
-function add(a, b) {
-    return a + b
+export async function getCityFeed(token) {
+  const response = await fetch(`https://api.waqi.info/feed/here/?token=${token}`);
```
Comment (rasimsen) [inline src/utils/loaders/getCityFeed.js:2]: url from the ENV file, please. 

Please note: we are gonna use Axios, we are gonna replace Axios next sprint

File: src/utils/loaders/getCityFeed.test.js (hunk 1 above)
Comment (Annel18) [reply src/utils/loaders/getCityFeed.test.js:17]: updated code and test to use axios

the whole URL is now from ENV file, not jus the token part (that was already from ENV file)

File: src/utils/loaders/getCityFeed.test.js (hunk 2 above)
Comment (Annel18) [reply src/utils/loaders/getCityFeed.test.js:25]: test suite updated

File: src/utils/loaders/getCityFeed.js (hunk 5 above)
Comment (Annel18) [reply src/utils/loaders/getCityFeed.js:2]: updated code and test to use axios

the whole URL is now from ENV file, not jus the token part (that was already from ENV file)

File: src/main.jsx [hunk 6]
```diff
@@ -8,12 +9,15 @@ import PageRealTime from './components/PageRealTime/index.js'
...
-// import { getCityFeed } from './utils/loaders/getCityFeed.js'
+import { getCityFeed } from './utils/loaders/getCityFeed.js'
 
 //! Styles
-import 'bootstrap/dist/css/bootstrap.min.css';
+import 'bootstrap/dist/css/bootstrap.min.css'
 import './styles/main.scss'
 
+//! Access environment variable
+const token = import.meta.env.VITE_TOKEN
```
[the remark this answers is not in the document]
Comment (Annel18) [reply src/main.jsx:19]: env.sample file added

**Pull request merged**